### PR TITLE
[channelz] modernize channel trace test

### PR DESCRIPTION
**Pull request by markdroth**
This cleans up the channel trace test to provide better output when tests fail, which will help diagnose the problem with b/339066599.

### CODE REVIEW

File: test/cpp/util/channel_trace_proto_helper.h [hunk 1]
```diff
@@ -19,17 +19,23 @@
...
 
-void ValidateChannelTraceProtoJsonTranslation(const char* json_c_str);
-void ValidateChannelProtoJsonTranslation(const char* json_c_str);
-void ValidateGetTopChannelsResponseProtoJsonTranslation(const char* json_c_str);
-void ValidateGetChannelResponseProtoJsonTranslation(const char* json_c_str);
-void ValidateGetServerResponseProtoJsonTranslation(const char* json_c_str);
-void ValidateSubchannelProtoJsonTranslation(const char* json_c_str);
-void ValidateServerProtoJsonTranslation(const char* json_c_str);
-void ValidateGetServersResponseProtoJsonTranslation(const char* json_c_str);
+void ValidateChannelTraceProtoJsonTranslation(const std::string& json_string);
```
Comment (drfloob) [inline test/cpp/util/channel_trace_proto_helper.h:27]: Why not use absl::string_view? Protobuf's `JsonStringToMessage` takes a string_view, so I think it should be a safe change.

File: test/core/channelz/channel_trace_test.cc [hunk 2]
```diff
@@ -43,82 +45,126 @@ namespace grpc_core {
...
+      IsJsonObject(::testing::ElementsAre(
+          ::testing::Pair("creationTimestamp", IsJsonString(::testing::_)),
+          ::testing::Pair("numEventsLogged",
+                          IsJsonStringNumber(num_events_logged_expected)))),
+      arg, result_listener);
+}
+
+void ValidateJsonProtoTranslation(const Json& json) {
+  std::string json_str = JsonDump(json);
+  grpc::testing::ValidateChannelTraceProtoJsonTranslation(json_str.c_str());
```
Comment (drfloob) [inline test/core/channelz/channel_trace_test.cc:166]: `.c_str()` is not needed.

**Review by drfloob — COMMENTED** [review]
Mostly looks good! It's much more readable now. I left two suggestions.

**Review by drfloob — APPROVED** [review]

File: test/cpp/util/channel_trace_proto_helper.h (hunk 1 above)
Comment (markdroth) [reply test/cpp/util/channel_trace_proto_helper.h]: Thanks for pointing this out!  I somehow thought that none of the protobuf APIs used absl yet, so I assumed that this needed to be `std::string`.

Done.

File: test/core/channelz/channel_trace_test.cc (hunk 2 above)
Comment (markdroth) [reply test/core/channelz/channel_trace_test.cc]: Done.

**Pull request closed without merging**